File: z/selection_sort.cc
class Solution {
 public:
  vector<int> solve(vector<int> array) {
    // write your solution here
    if (array.size() <= 1) {
      return array;
    }
    for (int i = 0; i < array.size(); ++i) {
      int min_index = i;
      for (int j = i + 1; j < array.size(); ++j) {
        if (array[min_index] > array[j]) {
          min_index = j;
        }
      }
      swap(array[i], array[min_index]);
    }
    return array;
  }
};

File: h/SelectionSort.java
public class Solution {
  public int[] solve(int[] array) {
    if (array.length <= 1) {
      return array;
    }

    for (int i = 0; i < array.length; i++) {
      int min = i;
      for (int j = i + 1; j < array.length; j++) {
        if (array[min] > array[j]) {
          min = j;
        }
      }
      swap(array, i, min);
    }
    return array;
  }

  private void swap(int[] array, int i, int min) {
    if (array[i] == array[min]) {
      return;
    }

    array[i] = array[i] + array[min];
    array[min] = array[i] - array[min];
    array[i] = array[i] - array[min];
  }
}
